packages/neon_http_client/test/client_test.dart

```diff
@@ -0,0 +1,170 @@
+import 'package:http/http.dart';
+import 'package:http/testing.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:neon_http_client/src/client.dart';
+import 'package:neon_http_client/src/interceptors/http_interceptor.dart';
+import 'package:test/test.dart';
+
+class _MockCookieStore extends Mock implements CookieStore {}
+
+class _MockInterceptor extends Mock implements HttpInterceptor {}
+
+class _FakeUri extends Fake implements Uri {}
+
+void main() {
+  final uri = Uri.parse('http://example.com');
+  Request fakeRequest() {
+    return Request('PUT', uri);
+  }
+
+  StreamedResponse fakeResponse() {
+    return StreamedResponse(const Stream.empty(), 200);
+  }
+
+  final mockedClient = MockClient((request) async {
+    return Response.fromStream(fakeResponse());
+  });
+  late NeonClient client;
+
+  setUpAll(() {
+    registerFallbackValue(_FakeUri());
+    registerFallbackValue(fakeResponse());
+    registerFallbackValue(fakeRequest());
+  });
+
+  tearDown(() {
+    client.close();
+  });
+
+  group(NeonClient, () {
+    test('adds user agent', () async {
+      client = NeonClient(
+        client: MockClient((request) async {
+          expect(
+            request.headers['user-agent'],
+            equals('NeonTest'),
+          );
+
+          return Response('', 200);
+        }),
+        userAgent: 'NeonTest',
+      );
+
+      await client.get(uri);
+    });
+
+    test('adds cookie store', () async {
+      final cookieStore = _MockCookieStore();
+      when(() => cookieStore.loadForRequest(any())).thenAnswer((_) => []);
+      when(() => cookieStore.saveFromResponse(any(), any())).thenAnswer((_) {});
+
+      client = NeonClient(
+        client: MockClient((request) async {
+          return Response('', 200);
+        }),
+        cookieStore: cookieStore,
+      );
+
+      await client.get(uri);
+
+      verify(() => cookieStore.loadForRequest(any(that: equals(uri)))).called(1);
+      verifyNever(() => cookieStore.saveFromResponse(any(that: equals(uri)), any()));
+    });
+
+    group('interceptors', () {
+      late HttpInterceptor interceptor;
+
+      setUp(() {
+        interceptor = _MockInterceptor();
+
+        client = NeonClient(
+          client: mockedClient,
+          interceptors: [interceptor],
+        );
+      });
+
+      test('does not intercept', () async {
+        when(() => interceptor.shouldInterceptRequest(any())).thenReturn(false);
+        when(() => interceptor.shouldInterceptResponse(any())).thenReturn(false);
+
+        final request = Request('GET', uri);
+        await client.send(request);
+
+        verifyNever(
+          () => interceptor.interceptRequest(
+            request: any(named: 'request', that: equals(request)),
+          ),
+        );
+        verifyNever(
+          () => interceptor.interceptResponse(
+            response: any(named: 'response', that: equals(fakeResponse)),
+            url: any(named: 'url', that: equals(uri)),
+          ),
+        );
```

Comment:
```suggestion
        verifyNever(
          () => interceptor.interceptRequest(
            request: any(named: 'request'),
          ),
        );
        verifyNever(
          () => interceptor.interceptResponse(
            response: any(named: 'response'),
            url: any(named: 'url'),
          ),
        );
```
Otherwise a mismatched value would cause the count to be 0 if there was something wrong (only necessary because this is a negative case)